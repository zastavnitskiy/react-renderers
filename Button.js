import React from 'react';

export default ({children}) => {
  return <button className="b-btn">
    <span className="b-btn__text">{children}</span>
  </button>
}