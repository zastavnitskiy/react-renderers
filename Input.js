import React from 'react';
import Button from './Button';

export default class Input extends React.Component {
  render() {
    return (<div className="b-input">
      <input className="b-input__input" defaultValue={this.props.defaultValue} />
      <Button>{this.props.buttonText}</Button>
    </div>)
  }
}