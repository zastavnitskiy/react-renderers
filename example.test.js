import React from "react";
import Button from "./Button";
import Input from "./Input"
import { render, mount, shallow } from "enzyme";

import toJson from "enzyme-to-json";
import TestRenderer from "react-test-renderer";
import ReactDOMServer from 'react-dom/server';
import prettier from "prettier";

const tests = [
    {
        name: "enzyme-static-renderer",
        render: render,
        serialize: wrapper => wrapper.html()
    },
    {
        name: "enzyme-mount-renderer",
        render: mount,
        serialize: wrapper => wrapper.html()
    },
    {
        name: "enzyme-shallow-renderer",
        render: shallow,
        serialize: wrapper => wrapper.html()
    },
    {
        name: "enzyme-static-renderer enzyme-to-json",
        render: render,
        serialize: wrapper => toJson(wrapper, { mode: "deep" })
    },
    {
        name: "enzyme-mount-renderer enzyme-to-json",
        render: mount,
        serialize: wrapper => toJson(wrapper, { mode: "deep" })
    },
    {
        name: "enzyme-shallow-renderer enzyme-to-json",
        render: shallow,
        serialize: wrapper => toJson(wrapper, { mode: "deep" })
    },
    {
        name: "react-test-renderer",
        render: TestRenderer.create,
        serialize: wrapper => wrapper
    },
    {
        name: "react-test-renderer toJSON",
        render: TestRenderer.create,
        serialize: wrapper => wrapper.toJSON()
    },
    {
        name: "react-test-renderer toTree",
        render: TestRenderer.create,
        serialize: wrapper => wrapper.toTree()
    },
    {
        name: "reactDOMServer.renderToStaticMarkup",
        render: ReactDOMServer.renderToStaticMarkup,
        serialize: wrapper => wrapper
    },
    {
        name: "reactDOMServer.renderToStaticMarkup + prettier",
        render: ReactDOMServer.renderToStaticMarkup,
        serialize: wrapper => prettier.format(wrapper)
    },
];

const examples = [
    {
        exampleName: "button",
        content: <Button>Hello</Button>
    },
    {
        exampleName: "input",
        content: (
            <div className="b-container">
                <Input defaultValue="Type here" buttonText="Send" />
            </div>
        )
    }
];

examples.forEach(({ exampleName, content }) => {
    tests.forEach(({ name, render, serialize }) => {
        test(`${exampleName} - ${name}`, () => {
            expect(serialize(render(content))).toMatchSnapshot();
        });
    });
});
