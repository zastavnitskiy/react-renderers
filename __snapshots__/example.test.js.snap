// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button - enzyme-mount-renderer 1`] = `"<button class=\\"b-btn\\"><span class=\\"b-btn__text\\">Hello</span></button>"`;

exports[`button - enzyme-mount-renderer enzyme-to-json 1`] = `
<button
  className="b-btn"
>
  <span
    className="b-btn__text"
  >
    Hello
  </span>
</button>
`;

exports[`button - enzyme-shallow-renderer 1`] = `"<button class=\\"b-btn\\"><span class=\\"b-btn__text\\">Hello</span></button>"`;

exports[`button - enzyme-shallow-renderer enzyme-to-json 1`] = `
<button
  className="b-btn"
>
  <span
    className="b-btn__text"
  >
    Hello
  </span>
</button>
`;

exports[`button - enzyme-static-renderer 1`] = `"<span class=\\"b-btn__text\\">Hello</span>"`;

exports[`button - enzyme-static-renderer enzyme-to-json 1`] = `
<button
  class="b-btn"
>
  <span
    class="b-btn__text"
  >
    Hello
  </span>
</button>
`;

exports[`button - react-test-renderer 1`] = `
<button
  className="b-btn"
>
  <span
    className="b-btn__text"
  >
    Hello
  </span>
</button>
`;

exports[`button - react-test-renderer toJSON 1`] = `
<button
  className="b-btn"
>
  <span
    className="b-btn__text"
  >
    Hello
  </span>
</button>
`;

exports[`button - reactDOMServer.renderToStaticMarkup + prettier 1`] = `
"<button class=\\"b-btn\\">
  <span class=\\"b-btn__text\\">Hello</span>
</button>;
"
`;

exports[`button - reactDOMServer.renderToStaticMarkup 1`] = `"<button class=\\"b-btn\\"><span class=\\"b-btn__text\\">Hello</span></button>"`;

exports[`input - enzyme-mount-renderer 1`] = `"<div class=\\"b-container\\"><div class=\\"b-input\\"><input class=\\"b-input__input\\" value=\\"Type here\\"><button class=\\"b-btn\\"><span class=\\"b-btn__text\\">Send</span></button></div></div>"`;

exports[`input - enzyme-mount-renderer enzyme-to-json 1`] = `
<div
  className="b-container"
>
  <div
    className="b-input"
  >
    <input
      className="b-input__input"
      defaultValue="Type here"
    />
    <button
      className="b-btn"
    >
      <span
        className="b-btn__text"
      >
        Send
      </span>
    </button>
  </div>
</div>
`;

exports[`input - enzyme-shallow-renderer 1`] = `"<div class=\\"b-container\\"><div class=\\"b-input\\"><input class=\\"b-input__input\\" value=\\"Type here\\"/><button class=\\"b-btn\\"><span class=\\"b-btn__text\\">Send</span></button></div></div>"`;

exports[`input - enzyme-shallow-renderer enzyme-to-json 1`] = `
<div
  className="b-container"
>
  <Input
    buttonText="Send"
    defaultValue="Type here"
  />
</div>
`;

exports[`input - enzyme-static-renderer 1`] = `"<div class=\\"b-input\\"><input class=\\"b-input__input\\" value=\\"Type here\\"><button class=\\"b-btn\\"><span class=\\"b-btn__text\\">Send</span></button></div>"`;

exports[`input - enzyme-static-renderer enzyme-to-json 1`] = `
<div
  class="b-container"
>
  <div
    class="b-input"
  >
    <input
      class="b-input__input"
      value="Type here"
    />
    <button
      class="b-btn"
    >
      <span
        class="b-btn__text"
      >
        Send
      </span>
    </button>
  </div>
</div>
`;

exports[`input - react-test-renderer 1`] = `
<div
  className="b-container"
>
  <div
    className="b-input"
  >
    <input
      className="b-input__input"
      defaultValue="Type here"
    />
    <button
      className="b-btn"
    >
      <span
        className="b-btn__text"
      >
        Send
      </span>
    </button>
  </div>
</div>
`;

exports[`input - react-test-renderer toJSON 1`] = `
<div
  className="b-container"
>
  <div
    className="b-input"
  >
    <input
      className="b-input__input"
      defaultValue="Type here"
    />
    <button
      className="b-btn"
    >
      <span
        className="b-btn__text"
      >
        Send
      </span>
    </button>
  </div>
</div>
`;

exports[`input - reactDOMServer.renderToStaticMarkup + prettier 1`] = `
"<div class=\\"b-container\\">
  <div class=\\"b-input\\">
    <input class=\\"b-input__input\\" value=\\"Type here\\" />
    <button class=\\"b-btn\\">
      <span class=\\"b-btn__text\\">Send</span>
    </button>
  </div>
</div>;
"
`;

exports[`input - reactDOMServer.renderToStaticMarkup 1`] = `"<div class=\\"b-container\\"><div class=\\"b-input\\"><input class=\\"b-input__input\\" value=\\"Type here\\"/><button class=\\"b-btn\\"><span class=\\"b-btn__text\\">Send</span></button></div></div>"`;
